reconcile/aus/base.py

```diff
@@ -739,6 +775,55 @@ def verify_lock_should_skip(
     return False
 
 
+def _create_upgrade_policy(
+    next_schedule: str, spec: ClusterUpgradeSpec, version: str
+) -> AbstractUpgradePolicy:
+    if spec.cluster.is_rosa_hypershift():
+        return ControlPlaneUpgradePolicy(
+            cluster=spec.cluster,
+            version=version,
+            schedule_type="manual",
+            next_run=next_schedule,
+        )
+    return ClusterUpgradePolicy(
+        cluster=spec.cluster,
+        version=version,
+        schedule_type="manual",
+        next_run=next_schedule,
+    )
+
+
+def _calulate_node_pool_diffs(
+    ocm_api: OCMBaseClient, spec: ClusterUpgradeSpec, now: datetime
+) -> Optional[UpgradePolicyHandler]:
+    node_pools = get_node_pools(ocm_api, spec.cluster.id)
+    if node_pools:
+        for pool in sorted(node_pools, key=lambda x: x["id"]):
+            current = get_node_pool_upgrade_policies(
+                ocm_api, spec.cluster.id, pool["id"]
+            )
+            if current:
+                continue
+            pool_version_id = pool.get("version", {}).get("id")
+            pool_version = get_version(ocm_api, pool_version_id)["raw_id"]
```

Comment:
can we avoid polling OCM continuously for this one? I guess we don't have the version `raw_id` directly in the node pool version, do we? Any caching possible?